test/specs/screenshot.js

```diff
@@ -2,70 +2,48 @@
 
 let expect = require('chai').expect
 let helper = require('../helper')
-let entities = require('entities')
 
 describe('Screenshots', () => {
+    beforeEach(helper.clean)
 
-  beforeEach(helper.clean)
+    it('can be taken during a test', () => {
+        return helper.run(['screenshot']).then((results) => {
+            expect(results).to.have.lengthOf(1)
+            expect(results[0]('test-case')).to.have.lengthOf(1)
 
-  it('can be taken in a "before all" task', () => {
-
-    return helper.run(['screenshot-before-all', 'screenshot']).then((code) => {
-
-      expect(code, 'wrong exit status code').to.equal(0)
-      return helper.getResultsXML()
-      
+            const screenshotFiles = helper.getResultFiles('png')
+            expect(screenshotFiles, 'no screenshot files attached').to.have.lengthOf(1)
+            expect(results[0]('test-case attachment[title="Screenshot"]')).to.have.lengthOf(1)
+        })
     })
-    .then((results) => {
 
-      expect(results).to.have.lengthOf(1)
-      expect(results[0]['ns2:test-suite']['test-cases']).to.have.lengthOf(1)
-      
-      let screenshotFiles = helper.getResultFiles(['jpg', 'png'])
-      expect(screenshotFiles, 'no screenshot files attached').to.have.lengthOf(1)
+    it('can take screenshot before each test and attach it accordingly', () => {
+        return helper.run(['screenshot-before-each']).then((results) => {
+            expect(results[0]('test-case')).to.have.lengthOf(2)
+            expect(results[0]('test-case attachment[title="Screenshot"]')).to.have.lengthOf(2)
 
+            const screenshotFiles = helper.getResultFiles('png')
+            expect(screenshotFiles, 'no screenshot files attached').to.have.lengthOf(2)
+        })
     })
 
-  })
-
-  it('can be taken in an "after all" task', () => {
-
-    return helper.run(['screenshot-after-all'], 'screenshot').then((code) => {
-
-      expect(code, 'wrong exit status code').to.equal(0)
-      return helper.getResultsXML()
-      
+    it('should not fail on uncaptured commands but write warning', () => {
+        return helper.run(['screenshot-before-all']).then((results) => {
+            const screenshotWarnings = helper.logs.warn.filter(log => {
+                return log.match(/GET \/wd\/hub\/session\/[^\/]*\/screenshot is not whithin any running test/)
+            })
+            expect(screenshotWarnings).to.have.lengthOf(1)
+            expect(helper.logs.error).to.have.lengthOf(0)
+            expect(results).to.have.lengthOf(1)
+        })
     })
-    .then((results) => {
 
-      expect(results).to.have.lengthOf(1)
-      expect(results[0]['ns2:test-suite']['test-cases']).to.have.lengthOf(1)
-      
-      let screenshotFiles = helper.getResultFiles(['jpg', 'png'])
-      expect(screenshotFiles, 'no screenshot files attached').to.have.lengthOf(1)
+    xit('[not supported yet] can be taken in an "before all" hook', () => {
```

Comment:
I have this test ignored because I couldn't find a simple solution for it.

Now there is a warning about command calls outside of running tests (It happens when you have some actions in before all). I tried to add them, but then you will always would have step about getting session, because it is also logged.

We need to find a way how to filter session commands in another way, rather than just matching with `/wd/hub/session` as well as it going now with screenshot logging